pages/user/edit.vue

```diff
@@ -0,0 +1,368 @@
+                  @click="removeSubject(subject)"
+                />
+              </div>
+              <div
+                class="rounded-full px-3 py-2 vx-row items-center text-ginger text-white mx-2 my-1"
+                style="background-color:#46C93A"
+              >
+                <i
+                  class="bx bx-plus-circle text-xl text-white mr-2"
+                  id="remove_icon"
+                  @click="activateModal()"
+                />
+
+                <!-- Modal for selecting un-selected subjects -->
+                <vs-dialog v-model="activeModel">
+                  <template #header>
+                    <h4 class="not-margin">
+                      <b>
+                        Add new
+                        Subejcts
+                      </b>
+                    </h4>
+                  </template>
+
+                  <div>
+                    <vs-checkbox
+                      :val="subject"
+                      v-for="(subject,index) in AllSubjectList"
+                      :key="index"
+                      v-model="UserInfo.interestedSubjects"
+                    >{{ subject }}</vs-checkbox>
+                  </div>
+                </vs-dialog>
+              </div>
+            </div>
+            <div class="w-full p-2 md:px-8" style>
+              <div
+                class="p-2 mb-2 rounded-lg border-solid border-gray-400 w-full text-2xl font-semibold"
+              >Biography</div>
+              <v-textarea filled v-model="UserInfo.bio"></v-textarea>
+              <vs-button
+                color="success"
+                type="filled"
+                size="large"
+                @click="saveChanges()"
+              >Save Changes</vs-button>
+            </div>
+          </div>
+        </div>
+      </VxCard>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { Component, Vue, Prop, mixins, Watch } from 'nuxt-property-decorator'
+import UserMixin from '~/mixins/UserMixin'
+import firestore from '~/plugins/firestore'
+
+import {
+  SubjectGroup_O,
+  Subject_O,
+  SubjectIconList,
+  SubjectOptions,
+  AllSubjectList
+} from '~/types/subjects'
+import NotesCard from '~/components/NotesCard.vue'
+import { Note_t, Note, Note_t_F } from '~/types/notes'
+import { School_O, SchoolList } from '~/types/schools'
+import SubjectsDropdown from '~/components/SubjectsDropdown.vue'
+import { firebaseAuth, GoogleAuthProvider } from '~/plugins/firebase'
+import { UserData } from '~/types/user'
+import { authStore } from '~/store'
+import ValidateImageMixin from '~/mixins/ValidateImageMixin'
+import functions from '~/plugins/firebaseFunctions'
+import storage from '~/plugins/firebaseStorage'
+
+@Component<UserProfile>({
+  components: {
+    NotesCard,
+    SubjectsDropdown
+  },
+  mounted() {
+    this.getUserNotes()
+    this.UserInfo = JSON.parse(JSON.stringify({ ...this.UserData }))
+  }
+})
+export default class UserProfile extends mixins(UserMixin, ValidateImageMixin) {
```

Comment:
The class name should be different - the Profile Page component is called UserProfile - call this one UserSettings or something